app/core/ports/service.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+from typing import BinaryIO, Protocol
+
+
+class FileService(ABC):
```

Comment:
lets make this a repository as this is also a data handler (simialr to sql but a file)

Reply:
```suggestion
class FileRepository(ABC):
```
Agreed, as discussed before.